src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -18,9 +18,28 @@ const buildDiagnose = (doc) => {
   return diagnose
 }
 
+const processAddedDiagnose = (diagnoses, doc) => {
+  const builtDiagnose = buildDiagnose(doc)
+  diagnoses.unshift({ builtDiagnose, id: doc.id })
+  return diagnoses
+}
+
+const processModifiedDiagnose = (diagnoses, doc) => {
+  const builtDiagnose = buildDiagnose(doc)
+  const index = diagnoses.findIndex(diagnose => diagnose.id === doc.id)
+  diagnoses[index] = { builtDiagnose, id: doc.id }
+  return diagnoses
+}
+
+const processRemovedDiagnose = (diagnoses, doc) => {
+  const index = diagnoses.findIndex(diagnose => diagnose.id === doc.id)
+  diagnoses.splice(index, 1)
+  return diagnoses
+}
```

Comment:
- Please add an empty line after constant declarations and before return statements
- These **functions are not pure**. They must not mutate the parameters. Create new arrays and return that (using spread operator to make a copy and `filter` to remove an item).
- Remember that for us, a modified diagnose is simply its deletion and immediate addition to the array. So, you can reuse the other two functions for that one.